File: 2nd Year/1st Semester/MAP/Laborator/Lab9/Lab9/model/SortingTask.cs
using Lab9.factory;

namespace Lab9.model;

public class SortingTask : Task
{
    private AbstractSorter _abstractSorter;
    private int[] _array;

    public SortingTask(string id, string description, int[] array, SortingStrategy sortingStrategy) : base(id, description)
    {
        _array = array;
        switch (sortingStrategy)
        {
            case SortingStrategy.Bubble:
                _abstractSorter = new BubbleSort();
                break;
            case SortingStrategy.Quick:
                _abstractSorter = new QuickSort();
                break;
            default:
                throw new ArgumentException("Invalid sorting strategy");
        }
    }

    public override void Execute()
    {
        _abstractSorter.Sort(_array);
        Console.WriteLine(string.Join(", ", _array) + "\n");

    }
}

File: 2nd Year/1st Semester/MAP/Laborator/SEM1_LAB2_MAP/model/SortingTask.java
package sem1_2.model;

import sem1_2.factory.SortingStrategy;

import java.util.Arrays;

public class SortingTask extends Task{

    private AbstractSorter abstractSorter;
    private final int[] array;

    public SortingTask(String id, String description, int[] array, SortingStrategy sortingStrategy) {
        super(id, description);
        this.array = array;

        switch (sortingStrategy) {
            case QUICK -> abstractSorter = new QuickSort();
            case BUBBLE -> abstractSorter = new BubbleSort();
        }
    }

    @Override
    public void execute() {
        abstractSorter.sort(array);
        System.out.println(Arrays.toString(array)+"\n");
    }
}
